src/WebJobs.Extensions.DurableTask/LifeCycleNotificationHelper.cs

```diff
@@ -0,0 +1,243 @@
+                hubName,
+                functionName,
+                version,
+                instanceId,
+                "",
+                OrchestrationRuntimeStatus.Completed);
+            await this.SendNotificationAsync(sendObject, hubName, functionName, version, instanceId, "", FunctionType.Orchestrator, isReplay, FunctionState.Completed);
+        }
+
+        public async Task OrchestratorFailedAsync(
+            string hubName,
+            string functionName,
+            string version,
+            string instanceId,
+            string reason,
+            FunctionType functionType,
+            bool isReplay)
+        {
+            if (!this.UseTrace)
+            {
+                return;
+            }
+
+            EventGridEvent[] sendObject = this.CreateEventGridEvent(
+                hubName,
+                functionName,
+                version,
+                instanceId,
+                reason,
+                OrchestrationRuntimeStatus.Failed);
+            await this.SendNotificationAsync(sendObject, hubName, functionName, version, instanceId, reason, FunctionType.Orchestrator, isReplay, FunctionState.Failed);
+        }
+
+        public async Task OrchestratorTerminatedAsync(
+            string hubName,
+            string functionName,
+            string version,
+            string instanceId,
+            string reason)
+        {
+            if (!this.UseTrace)
+            {
+                return;
+            }
+
+            EventGridEvent[] sendObject = this.CreateEventGridEvent(
+                hubName,
+                functionName,
+                version,
+                instanceId,
+                reason,
+                OrchestrationRuntimeStatus.Terminated);
+            await this.SendNotificationAsync(sendObject, hubName, functionName, version, instanceId, reason, FunctionType.Orchestrator, false, FunctionState.Terminated);
+        }
+
+        private EventGridEvent[] CreateEventGridEvent(
+            string hubName,
+            string functionName,
+            string version,
+            string instanceId,
+            string reason,
+            OrchestrationRuntimeStatus orchestrationRuntimeStatus)
+        {
+            return new[]
+            {
+                new EventGridEvent
+                {
+                    Id = Guid.NewGuid().ToString(),
+                    EventType = "orchestratorEvent",
+                    Subject = $"durable/orchestrator/{orchestrationRuntimeStatus}",
+                    EventTime = DateTime.UtcNow,
+                    Data = new
+                    {
+                        HubName = hubName,
+                        FunctionName = functionName,
+                        Version = version,
+                        InstanceId = instanceId,
+                        Reason = reason,
+                        EventType = orchestrationRuntimeStatus,
+                        EventTime = DateTime.UtcNow,
```

Comment:
Isn't this `EventTime` property redundant with the one defined by `EventGridEvent`?

Reply:
Thanks! Done!